### PR TITLE
Cross platform WinRM, PowerShell, and Multiple Machines

### PR DESCRIPTION
**Pull request by devigned**
This change set enables WinRM and PowerShell without the need to execute local PowerShell scripts. This simplifies the provisioning story as well as the remote script execution from command prompt scenarios. It also introduces `vagrant powershell [machine] [-c | s] [command | script file]`, which allows targeted execution of PowerShell scripts on the remote VM.

## things done
- add .ruby-version to gitignore
- lock gemfile to the minimum supported vagrant release
- patch azure so vhd names do not overlap (see: https://github.com/Azure/azure-sdk-for-ruby/pull/184)
- pin httpclient to the vagrant version for consistency
- add powershell and winrm functionality via a patch on top of the winrm communicator
- remove shell provisioner in favor of using default vagrant shell provisioner
- add functionality to discover vm guest os type based on vm image name
- add functionality to discover vm powershell endpoint
- added documentation for multi machine and windows configurations

### CODE REVIEW

File: lib/vagrant-azure/monkey_patch/winrm.rb [hunk 1]
```diff
@@ -0,0 +1,77 @@
+# patch no_ssl_peer_validation for ssl based winrm
+
+WinRM::HTTP::HttpTransport.class_eval do
+
+  # provide a patch for no peer verification.
+  # the patch exists upstream, but vagrant depends on an older version of winrm
```
Comment (jeffmendoza) [inline lib/vagrant-azure/monkey_patch/winrm.rb]: Which version?
Are all of these patches in newer upstream winrm?

Comment (devigned) [inline lib/vagrant-azure/monkey_patch/winrm.rb]: The no_ssl_peer_verification was introduced in winrm here https://github.com/WinRb/WinRM/commit/f4ece4384df2768bc7756c3c5c336db65b05c674 and is available in 1.2.0 and greater. The Vagrant embedded version of that gem is 1.1.3.

**Comment by jeffmendoza** [comment]
:+1:  LGTM, can you merge?

**Comment by devigned** [comment]
@jeffmendoza, will do.

**Pull request merged**